src/components/CMultiOptionButton.vue

```diff
@@ -0,0 +1,155 @@
+<template>
+  <v-menu :target="buttonGroup">
+    <template #activator="{ props: activatorProps }">
+      <v-btn-group
+        ref="buttonGroup"
+        block
+        :color="color"
+        :variant="variant"
+        :density="density"
+        rounded="xl"
+      >
+        <v-btn
+          :to="primaryButton.to"
+          :disabled="disabled"
+          :loading="loading"
+          @click="primaryButton.onClick"
+        >
+          {{ primaryButton.label }}
+        </v-btn>
+
+        <v-btn
+          v-bind="activatorProps"
+          :icon="mdiChevronDown"
+          :disabled="disabled"
+          class="pr-1 pl-0"
+          style="min-width: 35px; margin-left: 2px"
+        >
+        </v-btn>
+      </v-btn-group>
+    </template>
+
+    <v-list density="compact">
+      <v-list-item
+        v-for="(button, i) of additionalButtons"
+        :key="i"
+        :to="button.to"
+        :disabled="disabled"
+        @click="button.onClick"
+      >
+        <v-list-item-title> {{ button.label }} </v-list-item-title>
+      </v-list-item>
+    </v-list>
+  </v-menu>
+</template>
+
+<script setup lang="ts">
+import { mdiChevronDown } from "@mdi/js";
+import type { RouteLocationRaw } from "vue-router";
+import { computed, ref } from "vue";
+
+/**
+ * This ref is used to attach the menu to the whole button group.
+ */
+const buttonGroup = ref();
+
+/**
+ * Data of a button to display.
+ */
+export type ButtonData = {
+  /**
+   * The key of this button.
+   *
+   * This key should be unique since it is used to decide which button will be the primary button.
+   */
+  key: string;
+
+  /**
+   * The label of this button.
+   */
+  label: string;
+
+  /**
+   * Emitted when the user clicks on this button.
+   */
+  onClick?: () => {};
+
+  /**
+   * Denotes the target route of the link. You can find more information about the `to` prop on the vue-router documentation.
+   */
+  to?: RouteLocationRaw;
+};
+
+/**
+ * This component builds a button group with a default button and an arrow to reveal more options as alternatives to the default button.
+ */
+const props = withDefaults(
+  defineProps<{
+    /**
+     * Available buttons.
+     *
+     * Use `primaryButton` with the buttons key to decide which button will be chosen to be the primary button.
+     */
+    buttons: ButtonData[];
+
+    /**
+     * The key of the primary button to use.
+     *
+     * If the key does not exist, the first button will be used.
+     */
+    primaryButton: string;
```

Comment:
Do you think this is necessary? Is there a case where you can't just put the primary button in the first position of the button-array?

Reply:
I actually did this to simplify handling. You can define the buttons as in the story and simply select the primary button using the key. Otherwise you would have to sort the buttons manually via a ref or something like that